src/toolbarActions/toolbarFunctions.js

```diff
@@ -80,13 +80,34 @@ const saveAction = (state) => {
 };
 
 async function saveGraphMLFile(state) {
+    const { userAgent } = navigator;
```

Comment:
We really do not need this code block. The "Save" function works as-is. The bug #58 is intermittent and this code block does not address that issue.

Reply:
this block of code is used to detect browser ,  Because in case browser is FireFox we can't run window.showSaveFilePicker As it doesnot exist on that

Reply:
hey @pradeeban As firefox and safari not support window.showSaveFilePicker i wrote this code to detect browser and take the necessary action

Reply:
Aha, ok. Got it.

Can you send a new PR with just this change? I just want to make sure to merge one change at a time.

Reply:
I mean, avoid the below in this new PR:

    const handleSelectFile = (data) => {
        const fileExtensions = ['jpeg', 'jpg', 'png'];
        if (fileExtensions.includes(data.fileObj.name.split('.').pop())) {
            // eslint-disable-next-line no-alert
            alert('Wrong file extension');
            return;
        }



That is a separate PR that we will review.

Make sure to host your changes in your own concore editor deployment first. That way, I can test quicker too.

Reply:
this i separated as #89

Reply:
Yes, that is correct. Please submit a separate PR for this issue as well.